ncdu: Do not quit on Esc

unresolv opened the pull request:
#### What is the purpose of this change?

prevent quitting ncdu when pressing 'escape' or 'q' - dedicating that to closing the box

Ctrl-C will still quit



#### Was the change discussed in an issue or in the forum before?

<!--
Link issues and relevant forum posts here.
-->

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

fireonlive commented:
I'd like this; I often accidentally close ncdu after referencing the help pop-up. Sadly, it's also almost always with a remote that takes a long time to generate the listing and with lots of API requests that I may be charged for.

ncw commented:
It was the intention that `rclone ncdu` copies exactly the way `rclone ncdu` works.

For `ncdu`
- the key `q` and `ESC` closes the dialog. 
- the key `q` and `CTRL-C` quits.

For `rclone ncdu`
- the key `q` and `ESC` closes the dialog. 
- the key `q` and `ESC` and `CTRL-C` quits.

So maybe we should make the change that `ESC` no longer quits and encourage people to use that to close the dialog.

That would be `ncdu` compatible.

I would certainly miss `q` to close `rclone ncdu` as that is what I normally use!

What do you think?

unresolv commented:
@ncw thank you for your feedback!

that sounds good to me, and makes the diff a little smaller too :)

So now, for Ctrl-C, q, Esc, we:
1. Check if the box is there, or we are pressing Esc, if so, we close the box.
2. Otherwise, we exit

This means Esc will never close rclone ncdu, but q and Ctrl-C will work the same as before (close the box, then quit)

albertony commented:
Docs here need an update as well:

https://github.com/rclone/rclone/blob/2ab2ec29f9230f2959ae1e4428b07a4c7080aaed/cmd/ncdu/ncdu.go#L117

ncw reviewed (commented):
This is looking good now, but please add the docs as @albertony suggested - thank you :-)

unresolv commented:
@ncdu @albertony thank you, edited the documentation, as there was no entry for closing the box I added one for ESC, and removed ESC from quit

albertony commented:
> edited the documentation, as there was no entry for closing the box I added one for ESC, and removed ESC from quit

Perfect!

> @ncdu [...] thank you

Yes, I can join in with a big thanks to ncdu 😝

unresolv commented:
Oops I meant @ncw :’) 😊

On Wed, 10 Apr 2024 at 20:30, albertony <
***@***.***> wrote:

> edited the documentation, as there was no entry for closing the box I
> added one for ESC, and removed ESC from quit Perfect! @ncdu [...] thank you
> Yes, I can join in with a big thanks to ncdu 😝 —Reply
> *DuckDuckGo* removed one tracker. More
> <https://duckduckgo.com/-IerLo9B1uK3VCnEDDZw7cjfry97cCDSNLIJX-6zhuKsui47YiLKo9gOIdX22BbeHfrRVk0CEIcjPb3tduWQ1C22gxQtNWcjCs1MxkVbvwXOSIOVImARDYSAYyYj3S4gQ1aAs1fB84YtIsdicJGIUT73W_e6WvkLv6xy1wTnMfTyKm_XQBnyol7Qw2zzd-RNnCuF2-3eWLKKwCpSiN8WnA-DGlynB7Y7Ae9ynupYkIhHoZX-LuTnN8ABy9bs9vsNdFjluzwhxCvJIMhGdQA8BBNeJwZyTGFjq5TB5KHfdAHgq6zZNQ8qDDCbz0Yx4vlsvPZPhbzgcA5mairE2loaoyYw-26pRyxv4LvIIo_L8PEx3q5rMjZRaFzN5vEurb4ghS3f1ymJMQjlMunj1qbSzs2O_1l6MTHG_bUaxPbDAtsjco-oGv0aOCpfbZiHsX7L1O0zdFoVwLSdwXM-sjpn__CpI0Lst3lB9zMDk1tPjqK3EV5pAyWYMO>
> Report Spam
> <https://duckduckgo.com/-IerLo9B1uK3VCnEDDZw7cjfry97cCDSNLIJX-6zhuKsui47YiLKo9gOIdX22BbeHfrRVk0CEIcjPb3tduWQ1C22gxQtNWcjCs1MxkVbvwXOSIOVImARDYSAYyYj3S4gQ1aAs1fB84YtIsdicJGIUT73W_e6WvkLv6xy1wTnMfTyKm_XQBnyol7Qw2zzd-RNnCuF2-3eWLKKwCpSiN8WnA-DGlynB7Y7Ae9ynupYkIhHoZX-LuTnN8ABy9bs9vsNdFjluzwhxCvJIMhGdQA8BBNeJwZyTGFjq5TB5KHfdAHgq6zZNQ8qDDCbz0Yx4vlsvPZPhbzgcA5mairE2loaoyYw-26pRyxv4LvIIo_L8PEx3q5rMjZRaFzN5vEurb4ghS3f1ymJMQjlMunj1qbSzs2O_1l6MTHG_bUaxPbDAtsjco-oGv0aOCpfbZiHsX7L1O0zdFoVwLSdwXM-sjpn__CpI0Lst3lB9zMDk1tPjqK3EV5pAyWYMO>
>
> edited the documentation, as there was no entry for closing the box I
> added one for ESC, and removed ESC from quit
>
> Perfect!
>
> @ncdu [...] thank you
>
> Yes, I can join in with a big thanks to ncdu 😝
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rclone/rclone/pull/7731#issuecomment-2048200648>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAM553M46K4Y7ZWH4E6WSLDY4WAL5AVCNFSM6AAAAABFZXXEJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANBYGIYDANRUHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

unresolv commented:
thank you very much @ncw!
